Version is not supported

Hello on the newest Spigot Version (Minecraft 1.16.1) I get the version error. 
```
[13:33:05] [Server thread/ERROR]: Could not pass event InventoryClickEvent to Essentials v1.0-SNAPSHOT
org.bukkit.event.EventException: null
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:319) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at org.bukkit.plugin.RegisteredListener.callEvent(RegisteredListener.java:70) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at org.bukkit.plugin.SimplePluginManager.fireEvent(SimplePluginManager.java:589) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at org.bukkit.plugin.SimplePluginManager.callEvent(SimplePluginManager.java:576) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.PlayerConnection.a(PlayerConnection.java:2190) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.PacketPlayInWindowClick.a(SourceFile:32) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.PacketPlayInWindowClick.a(SourceFile:10) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.PlayerConnectionUtils.lambda$0(PlayerConnectionUtils.java:19) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.TickTask.run(SourceFile:18) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandler.executeTask(SourceFile:144) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandlerReentrant.executeTask(SourceFile:23) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandler.executeNext(SourceFile:118) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.MinecraftServer.aZ(MinecraftServer.java:944) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.MinecraftServer.executeNext(MinecraftServer.java:937) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.IAsyncTaskHandler.awaitTasks(SourceFile:127) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.MinecraftServer.sleepForTick(MinecraftServer.java:921) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.MinecraftServer.v(MinecraftServer.java:853) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at net.minecraft.server.v1_16_R1.MinecraftServer.lambda$0(MinecraftServer.java:165) ~[server.jar:git-Spigot-758abbe-3f0c333]
        at java.lang.Thread.run(Thread.java:748) [?:1.8.0_212]
Caused by: java.lang.ExceptionInInitializerError
        at net.wesjd.anvilgui.AnvilGUI$Builder.open(AnvilGUI.java:380) ~[?:?]
        at tk.therealjan.essentials.Events.HomesGUIEvent.onHomesGUIEvent(HomesGUIEvent.java:83) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_212]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_212]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[server.jar:git-Spigot-758abbe-3f0c333]
        ... 18 more
Caused by: java.lang.RuntimeException: Your server version isn't supported in AnvilGUI!
        at net.wesjd.anvilgui.version.VersionMatcher.lambda$match$1(VersionMatcher.java:49) ~[?:?]
        at java.util.Optional.orElseThrow(Optional.java:290) ~[?:1.8.0_212]
        at net.wesjd.anvilgui.version.VersionMatcher.match(VersionMatcher.java:49) ~[?:?]
        at net.wesjd.anvilgui.AnvilGUI.<clinit>(AnvilGUI.java:35) ~[?:?]
        at net.wesjd.anvilgui.AnvilGUI$Builder.open(AnvilGUI.java:380) ~[?:?]
        at tk.therealjan.essentials.Events.HomesGUIEvent.onHomesGUIEvent(HomesGUIEvent.java:83) ~[?:?]
        at sun.reflect.GeneratedMethodAccessor6.invoke(Unknown Source) ~[?:?]
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[?:1.8.0_212]
        at java.lang.reflect.Method.invoke(Method.java:498) ~[?:1.8.0_212]
        at org.bukkit.plugin.java.JavaPluginLoader$1.execute(JavaPluginLoader.java:315) ~[server.jar:git-Spigot-758abbe-3f0c333]
        ... 18 more
```

--- a/1_16_R1/pom.xml
+++ b/1_16_R1/pom.xml
@@ -1,0 +1,30 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>anvilgui-1_16_R1</artifactId>
+
+    <parent>
+        <groupId>net.wesjd</groupId>
+        <artifactId>anvilgui-parent</artifactId>
+        <version>1.3.0</version>
+    </parent>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.spigotmc</groupId>
+            <artifactId>spigot</artifactId>
+            <version>1.16.1-R0.1-SNAPSHOT</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>net.wesjd</groupId>
+            <artifactId>anvilgui-abstraction</artifactId>
+            <version>1.3.0</version>
+            <scope>compile</scope>
+        </dependency>
+    </dependencies>
+
+</project>

--- a/1_16_R1/src/main/java/net/wesjd/anvilgui/version/Wrapper1_16_R1.java
+++ b/1_16_R1/src/main/java/net/wesjd/anvilgui/version/Wrapper1_16_R1.java
@@ -1,0 +1,129 @@
+package net.wesjd.anvilgui.version;
+
+import net.minecraft.server.v1_16_R1.*;
+import org.bukkit.craftbukkit.v1_16_R1.CraftWorld;
+import org.bukkit.craftbukkit.v1_16_R1.entity.CraftPlayer;
+import org.bukkit.craftbukkit.v1_16_R1.event.CraftEventFactory;
+import org.bukkit.entity.Player;
+import org.bukkit.inventory.Inventory;
+
+public class Wrapper1_16_R1 implements VersionWrapper {
+    private int getRealNextContainerId(Player player) {
+        return toNMS(player).nextContainerCounter();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public int getNextContainerId(Player player, Object container) {
+        return ((AnvilContainer) container).getContainerId();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void handleInventoryCloseEvent(Player player) {
+        CraftEventFactory.handleInventoryCloseEvent(toNMS(player));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendPacketOpenWindow(Player player, int containerId, String guiTitle) {
+        toNMS(player).playerConnection.sendPacket(new PacketPlayOutOpenWindow(containerId, Containers.ANVIL, new ChatMessage(guiTitle)));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void sendPacketCloseWindow(Player player, int containerId) {
+        toNMS(player).playerConnection.sendPacket(new PacketPlayOutCloseWindow(containerId));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setActiveContainerDefault(Player player) {
+        toNMS(player).activeContainer = toNMS(player).defaultContainer;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setActiveContainer(Player player, Object container) {
+        toNMS(player).activeContainer = (Container) container;
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void setActiveContainerId(Object container, int containerId) {
+        //noop
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public void addActiveContainerSlotListener(Object container, Player player) {
+        ((Container) container).addSlotListener(toNMS(player));
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Inventory toBukkitInventory(Object container) {
+        return ((Container) container).getBukkitView().getTopInventory();
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    @Override
+    public Object newContainerAnvil(Player player, String guiTitle) {
+        return new AnvilContainer(player, guiTitle);
+    }
+
+    /**
+     * Turns a {@link Player} into an NMS one
+     *
+     * @param player The player to be converted
+     * @return the NMS EntityPlayer
+     */
+    private EntityPlayer toNMS(Player player) {
+        return ((CraftPlayer) player).getHandle();
+    }
+
+    /**
+     * Modifications to ContainerAnvil that makes it so you don't have to have xp to use this anvil
+     */
+    private class AnvilContainer extends ContainerAnvil {
+
+        public AnvilContainer(Player player, String guiTitle) {
+            super(getRealNextContainerId(player), ((CraftPlayer) player).getHandle().inventory,
+                    ContainerAccess.at(((CraftWorld) player.getWorld()).getHandle(), new BlockPosition(0, 0, 0)));
+            this.checkReachable = false;
+            setTitle(new ChatMessage(guiTitle));
+        }
+
+        @Override
+        public void e() {
+            super.e();
+            this.levelCost.set(0);
+        }
+
+        public int getContainerId() {
+            return windowId;
+        }
+
+    }
+
+}

--- a/api/pom.xml
+++ b/api/pom.xml
@@ -106,6 +106,11 @@
             <artifactId>anvilgui-1_15_R1</artifactId>
             <version>1.3.0</version>
         </dependency>
+        <dependency>
+            <groupId>net.wesjd</groupId>
+            <artifactId>anvilgui-1_16_R1</artifactId>
+            <version>1.3.0</version>
+        </dependency>
     </dependencies>
 
     <build>

--- a/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
+++ b/api/src/main/java/net/wesjd/anvilgui/version/VersionMatcher.java
@@ -33,7 +33,8 @@ public class VersionMatcher {
 			Wrapper1_13_R1.class,
 			Wrapper1_13_R2.class,
 			Wrapper1_14_R1.class,
-			Wrapper1_15_R1.class
+			Wrapper1_15_R1.class,
+			Wrapper1_16_R1.class
 	);
 
 	/**

--- a/pom.xml
+++ b/pom.xml
@@ -26,6 +26,7 @@
         <module>1_14_R1</module>
         <module>1_14_4_R1</module>
         <module>1_15_R1</module>
+        <module>1_16_R1</module>
     </modules>
 
     <repositories>

